android/src/main/java/com/oney/WebRTCModule/WebRTCModule.java

```diff
@@ -818,6 +1106,60 @@ public void onSetFailure(String s) {
         });
     }
 
+    private WritableMap mapForVideoCodecInfo(VideoCodecInfo info) {
+        WritableMap params = Arguments.createMap();
+        params.putString("mimeType", "video/" + info.name);
+        return params;
+    }
+
+    private WritableMap doReceiverGetCapabilities() {
+        VideoCodecInfo[] videoCodecInfos = mVideoDecoderFactory.getSupportedCodecs();
+        WritableMap params = Arguments.createMap();
+        WritableArray codecs = Arguments.createArray();
+        for(VideoCodecInfo codecInfo: videoCodecInfos) {
+            codecs.pushMap(mapForVideoCodecInfo(codecInfo));
+        }
+        params.putArray("codecs", codecs);
+        return params;
+    }
+
+    private WritableMap doSenderGetCapabilities() {
+        VideoCodecInfo[] videoCodecInfos = mVideoEncoderFactory.getSupportedCodecs();
+        WritableMap params = Arguments.createMap();
+        WritableArray codecs = Arguments.createArray();
+        for(VideoCodecInfo codecInfo: videoCodecInfos) {
+            codecs.pushMap(mapForVideoCodecInfo(codecInfo));
+        }
+        params.putArray("codecs", codecs);
+        return params;
+
+    }
+
+    @ReactMethod(isBlockingSynchronousMethod = true)
+    public WritableMap receiverGetCapabilities() {
```

Comment:
This was only called once, right?

Reply:
Yes, I made it a module constant on the typescript side that only gets called once, so it made sense to have it be blocking.